### ISSUE TITLE
a couple of issues

### ISSUE THREAD

**Issue by ctf0** [htmlhint/vscode-htmlhint#74]
- the `"htmlhint.documentSelector` wont work because of https://github.com/htmlhint/vscode-htmlhint/blob/07db8b265378ccf7f0064926abd8c2f1597f05fe/htmlhint/package.json#L32

- `htmlhint.configFile` isnt available https://github.com/htmlhint/vscode-htmlhint/blob/07db8b265378ccf7f0064926abd8c2f1597f05fe/htmlhint/package.json#L39

REF:
https://github.com/htmlhint/vscode-htmlhint/tree/main/htmlhint#settings

**Comment by coliff** [comment]
Thanks for reporting these issues! 
- On that first issue, is the solution to change the `activationEvents` to `onFileSystem:file` [like you did on your fork](https://github.com/ctf0/vscode-htmlhint/blob/master/htmlhint/package.json#L17)? Does that work as expected with no problems?
- For the second do you have a suggested fix for `htmlhint.configFile`?

**Comment by ctf0** [comment]
both of the issues are solved on my branch yes, as for does it work https://marketplace.visualstudio.com/items?itemName=ctf0.htmlhint the installation count says it all especially that thats all i've changed

**Comment by michael-nok** [comment]
@coliff Could you please include just the `activationEvents` changes provided by @ctf0 to allow for more startup options?

**Comment by rubensa** [comment]
Any work on this?  I'm trying to validate some html inside markdown documents.

I think that the:
```json
    "activationEvents": [
        "onFileSystem:file"
    ],
```
is needed so the extension can be activated for other files than HTML (so it can be used for markdown files if I specify `markdown` inside `htmlhint.documentSelector` configuration) .

The other change I can see is the addition of:
```json
                "htmlhint.documentSelector": {
                    "items": {
                        "type": "string"
                    }
                }
```
that I don't know if it is really necessary.

**Comment by rubensa** [comment]
@coliff For the `htmlhint.configFile` part @ctf0  added the `htmlhint.optionsFile` configuration.

In `htmlhint-server/src/server.ts` he added [`optionsFile`](https://github.com/microsoft/vscode-htmlhint/compare/main...ctf0:vscode-htmlhint:master#diff-ebd5d4df81281367ec41e0667be3b9ef594012b9faf46c494a318faac2f9d9fbR21) to the `Settings` interface:

```TypeScript
interface Settings {
    htmlhint: {
        enable: boolean;
        options: any;
        optionsFile: string;
    },
    [key: string]: any;
}
```

and the code to load the configuration if the option is specified inside [`getConfiguration`](https://github.com/microsoft/vscode-htmlhint/compare/main...ctf0:vscode-htmlhint:master#diff-ebd5d4df81281367ec41e0667be3b9ef594012b9faf46c494a318faac2f9d9fbR89):

```TypeScript
    else if (settings.htmlhint && settings.htmlhint.optionsFile) {
        options = loadConfigurationFile(settings.htmlhint.optionsFile)
    }
```

**Comment by rubensa** [comment]
@coliff Sorry to bother you but it's been more than a year since you said you where testing the change locally.  Any news on this?

**Comment by inosik** [comment]
Is there no other way to achieve what you intend other than `onFileSystem:file`? I'm trying to set up VS Code as my editor for Git and HTMLHint has an activation time of over 500 ms.

**Comment by coliff** [comment]
> Is there no other way to achieve what you intend other than `onFileSystem:file`? I'm trying to set up VS Code as my editor for Git and HTMLHint has an activation time of over 500 ms.

I don't think that change impacts activation time much. The latest version of the extension (1.0.5) has an activation time of 120ms on my Windows 11 laptop, and the first version (1.0.0) has an extension time of 110ms so it's nearly identical.

**Comment by inosik** [comment]
Just to be sure, are you talking about this value?

![Code_2024-09-26_08-10-24](https://github.com/user-attachments/assets/a47a16c3-8629-4bb9-a9aa-955cd15b5044)

This seems to be the sum of the time needed to load the extension and the time to actually activate it. The loading time is typically much higher for HTMLHint. Here's a snippet of the output of the "Developer: Startup Performance" command:

| Extension                                | Eager | Load Code | Call Activate | Finish Activate | Event                           | By                                       |
| ---------------------------------------- | ----- | --------- | ------------- | --------------- | ------------------------------- | ---------------------------------------- |
| HTMLHint.vscode-htmlhint                 | false | 671       | 30            | 0               | onFileSystem:file               | HTMLHint.vscode-htmlhint                 |

But I think I'll investigate this further when I have the time and come back to you if I have something actionable.

**Comment by coliff** [comment]
yes, but I have a very fast, relatively new Laptop which perhaps skews things somewhat 😅 
![image](https://github.com/user-attachments/assets/cc07bc4d-7335-4e0d-8d90-0dcbcfedf9b3)

**Comment by coliff** [comment]
Update: This has finally been added - it's in the latest version on VS Code Marketplace

![Image](https://github.com/user-attachments/assets/dc2245ac-591a-494b-a341-416c88789059)
